### DISCUSSION TITLE
knex postgis

### DISCUSSION THREAD

**Discussion by ymarcon** [feathersjs/feathers#3299] [Q&A] [answered]
Hi,

I would like to use the postgis flavor of postgres to make geo-aware queries, like finding entries within a bounding box. There is the [knex-postgis](https://github.com/jfgodoy/knex-postgis) extension that would do the job on the knex side. For feathers, I am wondering how I can extend/configure the built-in `KnexAdapter` to handle geo operations. I cannot figure out how the query schema can be used to add a new operator (such as for instance `$within`) and to add its translation into knex function calls. Also the `KnexAdapter.knexify()` seems to rely only on hard coded `METHODS` and `OPERATORS`.

Should I develop my own db adapter or there is a more elegant solution?

Thx

**Comment by daffl** [comment, marked as answer]
You can still [extend](https://feathersjs.com/guides/cli/service.class.html#overriding-methods) the `createQuery` or `knexify` methods and add your own functionality.

```ts
knexify(knexQuery: Knex.QueryBuilder, query: Query = {}, parentKey?: string): Knex.QueryBuilder {
  const q = super.knexify(knexQuery, query)

  return Object.keys(query).reduce((currentQuery, key) => {
    if (key === '$within') {
      const [lat, lon] = query[key]
      currentQuery.within(lat, lon)
    }
    return currentQuery
  })
}
```

**Reply by ymarcon** [reply]
Great, thanks, will try this.
